Tests/UnitTests/Core/Vertexing/IterativeVertexFinderTests.cpp

```diff
@@ -514,5 +518,105 @@ BOOST_AUTO_TEST_CASE(iterative_finder_test_user_track_type) {
   }
 }
 
+///
+/// @brief Unit test for IterativeVertexFinder with Athena reference data
+///
+BOOST_AUTO_TEST_CASE(iterative_finder_test_athena_reference) {
+  // Set up constant B-Field
+  ConstantBField bField(0.0, 0.0, 2_T);
+
+  // Set up Eigenstepper
+  EigenStepper<ConstantBField> stepper(bField);
+
+  // Set up propagator with void navigator
+  auto propagator = std::make_shared<Propagator>(stepper);
+
+  // Linearizer for BoundParameters type test
+  Linearizer::Config ltConfig(bField, propagator);
+  Linearizer linearizer(ltConfig);
+
+  using BilloirFitter = FullBilloirVertexFitter<BoundParameters, Linearizer>;
+
+  // Set up Billoir Vertex Fitter
+  BilloirFitter::Config vertexFitterCfg;
+
+  BilloirFitter bFitter(vertexFitterCfg);
+
+  // Impact point estimator
+  using IPEstimator = ImpactPointEstimator<BoundParameters, Propagator>;
+
+  IPEstimator::Config ipEstimatorCfg(bField, propagator);
+  IPEstimator ipEstimator(ipEstimatorCfg);
+
+  using ZScanSeedFinder = ZScanVertexFinder<BilloirFitter>;
+
+  static_assert(VertexFinderConcept<ZScanSeedFinder>,
+                "Vertex finder does not fulfill vertex finder concept.");
+
+  ZScanSeedFinder::Config seedFinderCfg(ipEstimator);
+
+  ZScanSeedFinder sFinder(seedFinderCfg);
+
+  // Vertex Finder
+  using VertexFinder = IterativeVertexFinder<BilloirFitter, ZScanSeedFinder>;
+
+  static_assert(VertexFinderConcept<VertexFinder>,
+                "Vertex finder does not fulfill vertex finder concept.");
+
+  VertexFinder::Config cfg(bFitter, linearizer, std::move(sFinder),
+                           ipEstimator);
+
+  cfg.useBeamConstraint = true;
+  cfg.maxVertices = 200;
+  cfg.maximumChi2cutForSeeding = 49;
+  cfg.significanceCutSeeding = 12;
+
+  VertexFinder finder(cfg);
+  VertexFinder::State state(magFieldContext);
+
+  auto csvData = readTracksAndVertexCSV(toolString);
+  auto tracks = std::get<TracksData>(csvData);
+
+  std::vector<const BoundParameters*> tracksPtr;
+  for (const auto& trk : tracks) {
+    tracksPtr.push_back(&trk);
+  }
+
+  VertexingOptions<BoundParameters> vertexingOptions(geoContext,
+                                                     magFieldContext);
+
+  vertexingOptions.vertexConstraint = std::get<BeamSpotData>(csvData);
+
+  // find vertices
+  auto findResult = finder.find(tracksPtr, vertexingOptions, state);
+
+  // BOOST_CHECK(findResult.ok());
+
+  if (!findResult.ok()) {
+    std::cout << findResult.error().message() << std::endl;
+  }
+
+  // Retrieve vertices found by vertex finder
+  // std::vector<Vertex<BoundParameters>> allVertices = *findResult;
+
+  // Test expected outcomes from athena implementation
```

Comment:
Can you remove the commented code?

Reply:
See my comment above to Roberts comment.

Reply:
Ok, understood.